moor_generator/lib/src/analyzer/moor/entity_handler.dart

```diff
@@ -56,6 +61,12 @@ class EntityHandler extends BaseAnalyzer {
 
         _lint(node, 'special @create table');
         entity.references.addAll(_findTables(node.statement));
+      } else if (entity is MoorView) {
+        final node =
+            _handleMoorDeclaration(entity, _views) as CreateViewStatement;
+        _lint(node, node.viewName);
+        entity.references.addAll(_findTables(node.query));
+        entity.references.addAll(_findViews(node.query));
```

Comment:
References are used to bring entities into the right order (so we ensure that nothing will get created before its dependencies are complete). So essentially, I think that other entity kinds would need to have support for views as well. For instance, if the definition of a trigger contains a select from a view, its `bodyReferences` should contain that view.

We might want to introduce a shared common class for `MoorTable` and `MoorView` (maybe something like `MoorEntityWithResultSet`) and then have a single method for that.

Reply:
Yeah, I was thinking the same.